lib/trie/database.go

```diff
@@ -179,6 +194,21 @@ func (t *Trie) load(db chaindb.Database, curr Node) error {
 		}
 	}
 
+	l, ok := curr.(*node.Leaf)
+	if ok {
+		// see if there is a child trie with this hash
+		childTrie := NewEmptyTrie()
+		err := childTrie.Load(db, common.NewHash(l.Value))
+		if err != nil {
+			// TODO: Avoid only when state.ErrTrieDoesNotExist
+		} else {
+			if err = t.PutChild(l.Value, childTrie); err != nil {
+				return err
+			}
+		}
+
+	}
```

Comment:
we shouldn't do this for all leaf nodes, but we should instead be checking the sub-trie that has the key `ChildStorageKeyPrefix`, as all leaf nodes that contain `ChildStorageKeyPrefix` in them will point to a child trie

Reply:
addressed